bolt-modules/log/lib/puppet/functions/log/debug.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'bolt/util/format'
+
+# Log a debugging message.
+#
+# Messages logged at this level typically include detailed information about
+# what a plan is doing. For example, you might log a message at the `debug`
+# level that shows what value is returned from a function invocation.
+#
+# See [Logs](logs.md) for more information about Bolt's log levels.
+#
+# > **Note:** Not available in apply block
+Puppet::Functions.create_function(:'log::debug') do
+  # Log a debugging message.
+  # @param message The message to log.
+  # @example Log a debugging message
+  #   log::trace("Function frogsay returned: ${result}")
+  dispatch :log_debug do
+    param 'Any', :message
+    return_type 'Undef'
+  end
+
+  def log_debug(message)
+    unless Puppet[:tasks]
+      raise Puppet::ParseErrorWithIssue.from_issue_and_stack(
+        Bolt::PAL::Issues::PLAN_OPERATION_NOT_SUPPORTED_WHEN_COMPILING,
+        action: 'log::debug'
+      )
+    end
+
+    Puppet.lookup(:bolt_executor).tap do |executor|
+      executor.report_function_call(self.class.name)
+      executor.publish_event(type: :log, level: :debug, message: Bolt::Util::Format.stringify(message))
```

Comment:
Hm, I think a more specific event type like `plan_function_log` might be helpful if we want to add more types of events in the future. `plan_log` might work but is still kind of ambiguous...

Reply:
I'm not sure the extra specificity is needed? All the event does is send the message directly to the logger; it doesn't really care or need to know that the message came from a plan function. If we were doing something like sending analytics that the log came from a plan function, then I can see the extra specificity making sense.

Reply:
Hm, yeah, I also realized we can just update this type if we want something else to have `log` type in the future.